internal/pkg/manifest/svc.go

```diff
@@ -495,6 +495,46 @@ func (rr RoutingRuleConfiguration) exposedPorts(exposedPorts []ExposedPort, work
 	}
 }
 
+// exportPorts returns any new ports that should be exposed given the network load balancer
+// configuration that's not part of the existing containerPorts.
+func (cfg NetworkLoadBalancerConfiguration) exposedPorts(exposedPorts []ExposedPort, workloadName string) ([]ExposedPort, error) {
```

Comment:
💭  I wonder if this ⬇️ can simplify the logic a little... It looks a bit tricky that we are appending and passing `exposedPorts` as a parameter around.
```go
// exportPorts returns any new ports that should be exposed given the application load balancer configuration.
func (rr ALBConfig) exposedPorts() ([]ExposedPort, error)

// exportPorts returns any new ports that should be exposed given the network load balancer configuration.
func (cfg NLBConfig) exposedPorts() ([]ExposedPort, error)

// exportPorts returns any new ports that should be exposed given the image configuration.
func (cfg ImageWithPortAndHealthcheck) exposedPorts(workloadName string) []ExposedPort

// exportPorts returns any new ports that should be exposed given the image configuration.
func (cfg ImageWithHealthcheckAndOptionalPort) exposedPorts(workloadName string) []ExposedPort

// ExposedPorts returns all the ports that are container ports available to receive traffic.
func (b *BackendService) ExposedPorts() ([]ExposedPort, error) {
	var exposedPorts []ExposedPort
	workloadName := aws.StringValue(b.Name)
	exposedPorts = append(exposedPorts, b.ImageConfig.exposedPorts(workloadName)...)
	for name, sidecar := range b.Sidecars {
		out, err := sidecar.exposedPorts(name)
		if err != nil {
			return nil, err
		}
		exposedPorts = append(exposedPorts, out...)
	}
	exposedPorts = append(exposedPorts, b.RoutingRule.exposedPorts(workloadName)...)
	return sortExposedPorts(exposedPorts), nil
}

func sortExposedPorts(ports []ExposedPorts) []ExposedPort {
   // Remove repeatedly exposed ports.
   // Sort
}
```

Reply:
I had the exact same logic written before but the only problem with this logic is sortExposedPorts doesn't know which duplicate port to remove out of two ports. 

Let me take an example manifest
```
name: lbservice
type: Load Balanced Web Service

image:
  port: 80

http:
  path: '/'
  target_port: 81
  
sidecars:
  nginx:
    image:
      port: 81
```

In the following manifest we should expose following ports 
lbservice - 80
nginx - 81

But with the logic mentioned above we will get following exposed ports 
lbservice - 80
lbservice - 81
nginx - 81

And now we do not know which duplicate port to be removed from the list? Does that make sense?